src/language-client.ts

```diff
@@ -239,8 +240,19 @@ export class LanguageClient implements Disposable {
       return
     }
 
+    const lspDiffWithTimeout = (oldCode: string, newCode: string) => {
```

Comment:
maybe a comment to explain why would be nice ?

Reply:
Sure!

Reply:
I squashed it

Reply:
Is it clear enough?

Reply:
or maybe, do you have a better idea to fix this issue?

Reply:
Perfectly clear (and interesting), thanks !
I don't think I have a better idea, maybe an option would be to apply a fast algorithm not to compute diff but to detect if it's worth to computing it (some kind of levenshtein distance equivalent ?).

Reply:
ok then this seems to be a good solution !

Reply:
What is not ideal in the current approach is it makes the code nodejs-dependant, even if it's probably not a really big deal

Reply:
Yes I don't think that's an issuer

Reply:
Also, I'm not sure about what timeout to set:
- A simple normal diff (inserting or deleting some chars) is computed in less than a ms on my machine
- Copy/pasting ~2000 line of java over the default coderpad java code leads to a ~1.2seconds diff computation

Reply:
20ms (maybe 50?) seems reasonable, we could update it if necessary. What is the risk to return the newCode instead of the diff ?

Reply:
The only risk is the LSP reloading the file for nothing